lib/split/algorithms/systematic_sampling.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+module Split
+  module Algorithms
+    module SystematicSampling
+      def self.choose_alternative(experiment)
+        count = experiment.next_cohorting_block_index
+
+        block_length = experiment.cohorting_block_magnitude * experiment.alternatives.length
+        block_num, index = count.divmod block_length
+
+        r = Random.new(block_num + experiment.cohorting_block_seed)
+        block = (experiment.alternatives*experiment.cohorting_block_magnitude).shuffle(random: r)
+        block[index]
```

Comment:
What we do here to create any block:

take the list of alternatives:

`['red', 'blue', 'green']`

repeat it `cohorting_block_magnitude` times:

`['red', 'blue', 'green', 'red', 'blue', 'green', 'red', 'blue', 'green']`

shuffle it (seeded by `cohorting_block_seed` and the current block #)

`['green', 'red', 'blue', 'red', 'blue', 'green', 'red', 'green', 'blue']`